.github/workflows/test_rocprim.yml

```
@@ -0,0 +1,92 @@
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: rocPRIM test
+
+on:
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      amdgpu_target:
+        type: string
+        default: gfx94X-dcgpu
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      amdgpu_target:
+        type: string
+        default: gfx94X-dcgpu
+  push:
+    branches:
+      - ADHOCBUILD
+
+jobs:
+  test_rocprim:
+    name: "Test rocPRIM"
+    runs-on: linux-mi300-1gpu-ossci-rocm
+    strategy:
+      fail-fast: true
+    defaults:
+      run:
+        shell: bash
+    env:
+      VENV_DIR: ${{ github.workspace }}/.venv
+      ARTIFACT_RUN_ID: "${{ inputs.artifact_run_id != '' && inputs.artifact_run_id || github.run_id }}"
+      BUILD_ARTIFACTS_DIR: "build/artifacts"
+
+    # MOSTLY BOILER PLATE ABOVE.
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Install the AWS tool
+        run: ./dockerfiles/cpubuilder/install_awscli.sh
+
+      - name: "Setting up Python"
+        id: setup_python
+        uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+        with:
+          python-version: 3.11
+
+      - name: Create Python venv
+        run: |
+          python -m venv ${VENV_DIR}
+          source ${VENV_DIR}/bin/activate
+          pip install -r requirements-test.txt
+
+      - name: Download and Unpack Artifacts
+        run: |
+          source ${VENV_DIR}/bin/activate
+
+          # TODO(sosa): We should infer this from the runner. For Libs and Above it should be based on the
+          # runner, everything else is generic. Use just generic since this is all core/below.
+          VARIANT="_generic"
+          TARGET_VARIANT="_${{ inputs.amdgpu_target }}"
+          echo "Making ${BUILD_ARTIFACTS_DIR} directory"
+          mkdir -p "${BUILD_ARTIFACTS_DIR}"
+
+          echo "Downloading artifacts"
+          aws s3 cp s3://therock-artifacts/${ARTIFACT_RUN_ID}/sysdeps_lib"${VARIANT}".tar.xz "${BUILD_ARTIFACTS_DIR}" --no-sign-request
+          aws s3 cp s3://therock-artifacts/${ARTIFACT_RUN_ID}/base_lib"${VARIANT}".tar.xz "${BUILD_ARTIFACTS_DIR}" --no-sign-request
+          aws s3 cp s3://therock-artifacts/${ARTIFACT_RUN_ID}/amd-llvm_lib"${VARIANT}".tar.xz "${BUILD_ARTIFACTS_DIR}" --no-sign-request
+          aws s3 cp s3://therock-artifacts/${ARTIFACT_RUN_ID}/prim_test"${TARGET_VARIANT}".tar.xz "${BUILD_ARTIFACTS_DIR}" --no-sign-request
+
+          echo "Unpacking artifacts"
+          pushd "${BUILD_ARTIFACTS_DIR}"
+          mkdir output_dir
+          python ${{ github.workspace }}/build_tools/fileset_tool.py artifact-flatten *.tar.xz -o output_dir --verbose
+          sudo rsync --archive --keep-dirlinks --verbose output_dir/* /
+          popd
```

nit: I think the `pushd` and `popd` actually detract from readability here. You could also use `cd` since each step starts at the same working directory.

Not sure about the rsync command... maybe this?

```suggestion
          mkdir ${BUILD_ARTIFACTS_DIR}/output_dir
          python build_tools/fileset_tool.py artifact-flatten *.tar.xz -o ${BUILD_ARTIFACTS_DIR}/output_dir --verbose
          sudo rsync --archive --keep-dirlinks --verbose ${BUILD_ARTIFACTS_DIR}/output_dir/* ${BUILD_ARTIFACTS_DIR}
```